pyfr/readers/cgns.py

```diff
@@ -0,0 +1,433 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+from ctypes import (POINTER, RTLD_GLOBAL, create_string_buffer,
+                    c_char_p, c_int, c_void_p)
+import re
+
+import numpy as np
+
+from pyfr.ctypesutil import load_library
+from pyfr.readers import BaseMeshConverter, BaseReader
+from pyfr.readers.nodemaps import CGNSNodeMaps
+from pyfr.util import setenv
+
+
+class CGNSWrappers(object):
+    def __init__(self):
+        # Load CGNS
+        try:
```

Comment:
Lets require the new version of CGNS.  Solves a lot of issues.  Although the code for loading HDF5 is clean it still exposes people to things such as RTLD_GLOBAL and will become harder to test as time goes on (as more of us upgrade our CGNS versions).  Drop the HDF5 loading code, let it through an exception, and we can put a note in the documentation stating the required version.